crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -575,39 +575,55 @@ impl Room {
                 }
             }
 
-            (names, None)
+            (names, None, None)
         } else {
             let mut joined_members = self.members(RoomMemberships::JOIN).await?;
 
             // Make the ordering deterministic.
             joined_members.sort_unstable_by(|lhs, rhs| lhs.name().cmp(rhs.name()));
 
-            // We can make a good prediction of the total number of members here. This might
-            // be incorrect if the database info is outdated.
+            // We can make a good prediction of the total number of joined members here.
+            // This might be incorrect if the database info is outdated.
             let num_joined = Some(joined_members.len());
 
-            (
-                joined_members
-                    .into_iter()
-                    .take(NUM_HEROES)
-                    .filter_map(|u| (u.user_id() != own_user_id).then(|| u.name().to_owned()))
-                    .collect(),
-                num_joined,
-            )
+            let mut invited_members = self.members(RoomMemberships::INVITE).await?;
```

Comment:
Can you avoid the overhead of this extra `self.members()` call by passing a different filter to the call at line 580?

Reply:
Done. Since we don't really care for the separate count of joined and invited, I ended up using the total everywhere. I also split the code into a new method.